src/main/java/core/basesyntax/service/DataProcess.java

```diff
@@ -0,0 +1,9 @@
+package core.basesyntax.service;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.strategy.OperationStrategy;
+import java.util.List;
+
+public interface DataProcess {
+    void addDataToDB(List<FruitTransaction> fruitTransactions, OperationStrategy operationStrategy);
```

Comment:
It's not a mistake, but I'd better move operationStrategy from the method parameter and put as a field on the class level. Since operationStrategy probably doesn't need to be changed on a specific instance it's better to just pass it into constructor, or inject it via Injector.
The problem that your approach may cause is that if addDataToDB method is used in a lot of places, you would need to pass operationStrategy every time you call it.